File: edukate-notifier/src/main/java/io/github/sanyavertolet/edukate/notifier/dtos/SimpleNotificationDto.java
package io.github.sanyavertolet.edukate.notifier.dtos;

import com.fasterxml.jackson.annotation.JsonTypeName;
import lombok.*;

import java.time.Instant;

@Getter
@Setter
@JsonTypeName("simple")
public final class SimpleNotificationDto extends BaseNotificationDto {
    private String title;
    private String message;
    private String source;

    public SimpleNotificationDto(
            String uuid, Boolean isRead, Instant createdAt,
            String title, String message, String source
    ) {
        super(uuid, isRead, createdAt);
        this.title = title;
        this.message = message;
        this.source = source;
    }
}

File: edukate-notifier/src/main/kotlin/io/github/sanyavertolet/edukate/notifier/dtos/SimpleNotificationDto.kt
package io.github.sanyavertolet.edukate.notifier.dtos

import com.fasterxml.jackson.annotation.JsonTypeName
import java.time.Instant

@JsonTypeName("simple")
data class SimpleNotificationDto(
    override val uuid: String,
    override val isRead: Boolean,
    override val createdAt: Instant,
    val title: String,
    val message: String,
    val source: String,
) : BaseNotificationDto
